src/sentry/api/serializers/rest_framework/notification_action.py

```diff
@@ -38,21 +39,57 @@ class NotificationActionInputData(TypedDict):
     target_display: str
 
 
+@extend_schema_serializer(exclude_fields=["sentry_app_id", "target_type"])
 class NotificationActionSerializer(CamelSnakeModelSerializer):
     """
     Django Rest Framework serializer for incoming NotificationAction API payloads
     """
 
-    integration_id = serializers.IntegerField(required=False)
-    sentry_app_id = serializers.IntegerField(required=False)
-    projects = serializers.ListField(child=ProjectField(scope="project:read"), required=False)
-
-    service_type = serializers.CharField()
-    target_type = serializers.CharField()
-    trigger_type = serializers.CharField()
-
-    target_identifier = serializers.CharField(required=False)
-    target_display = serializers.CharField(required=False)
+    integration_id = serializers.IntegerField(
+        help_text="""ID of the integration used as the notification service. For example, this would
+            be the Slack Integration ID if the action has Slack notify organization members.
+            This field is required if **service_type** is `slack`, `pagerduty` or `opsgenie`.""",
+        required=False,
+    )
+
+    # Optional and not needed for spike protection so not documenting
+    # TODO: Include in documentation when any notification action works with sentry_app_id
+    sentry_app_id = serializers.IntegerField(
+        required=False,
+    )
+    projects = serializers.ListField(
+        help_text="""List of projects slugs that Notificaton Action is created for""",
+        child=ProjectField(scope="project:read"),
+        required=False,
+    )
+
+    service_type = serializers.CharField(
+        help_text="Service that is used for sending the notification\n"
+        + """- `email`\n"""
+        + """- `slack`\n"""
+        + """- `sentry_notification`\n"""
+        + """- `pagerduty`\n"""
+        + """- `opsgenie`\n"""
+    )
+
+    # TODO: Include in documentation when any notification action works with anything other than "specific"
+    target_type = serializers.CharField(
+        required=False,
+        default="specific",
```

Comment:
@leeandher since this was not used anywhere else, I made it default to specific and removed it from docs. Let me know what you think.